Wrappers/Python/test/test_DataContainer.py

```diff
@@ -571,48 +571,166 @@ def test_AcquisitionGeometry_allocate(self):
         self.assertEqual(order[1], sino.dimension_labels[1])
         self.assertEqual(order[2], sino.dimension_labels[2])
         self.assertEqual(order[2], sino.dimension_labels[2])
-        
-
-        
+                
         try:
             z = AcquisitionData(numpy.random.randint(10, size=(2,3)), geometry=ageometry)
             self.assertTrue(False)
         except ValueError as ve:
             print (ve)
             self.assertTrue(True)
+
+    def test_ImageGeometry_allocate_dtype(self):
+        
+        print("Test ImageGeometry dtype\n")
+        ig = ImageGeometry(3,3)
+        print("The default dtype of the ImageGeometry is {}".format(ig.dtype))
+        self.assertEqual(ig.dtype, numpy.float32)
+        
+        bg = BlockGeometry(ig,ig)
+        print("The default dtype of the BlockImageGeometry is {}".format(bg.dtype))   
+        self.assertEqual(bg.dtype, tuple([numpy.float32]*len(ig.shape)))
+         
+        print("Change it to complex")
+        ig.dtype = numpy.complex
+        print("The ImageGeometry dtype is now {} ".format(ig.dtype))         
+        self.assertEqual(ig.dtype, numpy.complex)
+
+        print("Test ImageGeometry allocate")
+        data = ig.allocate()
+        print("Data dtype is now {} ".format(ig.dtype))  
+        self.assertEqual(data.dtype, ig.dtype)
+        print("Data geometry dtype is now")
+        print(data.geometry.dtype)
+        self.assertEqual(data.geometry.dtype, ig.dtype) 
+
+        print("Allocate data with different dtype, e.g: numpy.int64 from the same ImageGeometry")
+        data = ig.allocate(dtype=numpy.int64)
+        self.assertEqual(data.dtype, numpy.int64) 
+        print("Data dtype is now {}".format(data.dtype))
+        print("Data geometry dtype is now {}".format(data.geometry.dtype))
+        self.assertEqual(data.geometry.dtype, numpy.int64)
+        self.assertEqual(data.dtype, numpy.int64)
+
+        print("The dtype of the ImageGeometry remain unchanged ig.dtype =  {}".format(ig.dtype))
+        self.assertEqual(ig.dtype, numpy.complex)  
+
+    def test_AcquisitionGeometry_allocate_dtype(self):
```

Comment:
This test and `test_ImageGeometry_allocate_dtype` and `test_VectorGeometry_allocate_dtype` are exactly the same code testing 3 different classes. For maintainability, I suggest to create one method in the test class as

```python

def dtype_allocate_test(self, geometry):
    classname = geometry.__class__.__name__
    # print("The default dtype of the {} is {}".format(classname , geometry.dtype))
    self.assertEqual(geometry.dtype, numpy.float32)

    print("Change it to complex")
    geometry.dtype = numpy.complex
    print("The {}dtype is now {} ".format(classname , geometry.dtype))         
    self.assertEqual(geometry.dtype, numpy.complex)

    print("Test {} allocate".format(classname ))
    data = geometry.allocate()
    print("Data dtype is now {} ".format(geometry.dtype))  
    self.assertEqual(data.dtype, geometry.dtype)
    print("Data geometry dtype is now")
    print(data.geometry.dtype)
    self.assertEqual(data.geometry.dtype, geometry.dtype) 

    print("Allocate data with different dtype, e.g: numpy.int64 from the same {}".format(classname ))
    data = geometry.allocate(dtype=numpy.int64)
    self.assertEqual(data.dtype, numpy.int64) 
    print("Data dtype is now {}".format(data.dtype))
    print("Data geometry dtype is now {}".format(data.geometry.dtype))
    self.assertEqual(data.geometry.dtype, numpy.int64)
    self.assertEqual(data.dtype, numpy.int64)

    print("The dtype of the {}remain unchanged ig.dtype =  {}".format(classname, geometry.dtype))
    self.assertEqual(geometry.dtype, numpy.complex)  
```

Then you can create the appropriate test methods as

```python

def test_AcquisitionGeometry_allocate_dtype(self):
    # Detectors
    detectors =  10

    # Angles
    angles = numpy.linspace(0,180,180, dtype='float32')

    # Setup acquisition geometry
    ag = AcquisitionGeometry.create_Parallel2D()\
                                .set_angles(angles)\
                                .set_panel(detectors, pixel_size=0.1)   
    # run the tests
    self.dtype_allocate_test(ag)

def test_ImageGeometry_allocate_dtype(self):
    ig = ImageGeometry(3,3)
    # run the tests
    self.dtype_allocate_test(ig)

def test_VectorGeometry_allocate_dtype(self):
    ig = VectorGeometry(3)
    # run the tests
    self.dtype_allocate_test(ig)
```

The 3 tests will all run the same test code in `dtype_allocate_test`. In case of a change this will transfer immediately to the 3 classes.